Follow up PR for Issue #552: Contribution guidelines as per TF structure

ricky3dec opened the pull request:
1. Add new file CONTRIBUTING.md
2. Add new file overview.md
3. Update README.md
4. Update the Table of contents .yaml (_toc.yaml)

ricky3dec commented:
Hi @yongtang , please review the changes and let me know if I need to pull up or out any sections from the guides.

yongtang commented:
@MarkDaoust we are adding an overview page to TOC. Wondering if you could take a look and see if it fits tensorflow.org's style?

lamberta commented:
@ricky3dec I think a single CONTRIBUTING.md file in the root directory is best. Pages in the /docs directory will get published to the website, and we usually don't publish project contributing guidelines--we just link off to the doc in GitHub. (Though we do link to https://www.tensorflow.org/community/contribute)

lamberta commented:
Though, looking a bit further, your CONTRIBUTING.md doc contains more technical detail than typically in these files. I would perhaps move this info out into a separate guide called setup.md (testing.md, or something like that), *if* you want end-users to read it. If it's just for project contributors, I'd keep it as a GitHub-only doc.

ricky3dec commented:
Thank you @lamberta for your inputs.
I can remove the contributing.md from /docs directory, and would just keep it in root directory.

As you suggested, I am planning to take out some of technical details from existing contributing.md and put them into setup.md and style_guide.md. Also would retain the original contents of root readme.md, and once all the content is in right place, we can trim the root readme.md as well. Hope this sounds ok?

ricky3dec commented:
@lamberta and @yongtang , I have made above mentioned changes in this new PR.
* Trimmed down contents of contributing.md and removed technical details
* Added naming conventions and code style details to style_guide.md
* Retained the contents of readme.md and added link to contributing.md.

Going forward we can trim down contents of main readme.md as well.

Please let me know, if any other changes are required. Thanks

yongtang commented:
@MarkDaoust @lamberta the PR looks good to me. Can you take a look as well, since it updates `docs` directory so I assume changed could be populated to tensorflow.org/io ?

lamberta commented on CONTRIBUTING.md:
````diff
@@ -1,178 +1,32 @@
-  ...
-  return core_ops.io_read_avro(...);
-```
+Please see our [Style Guide](STYLE_GUIDE.md) for more details.
````
We have a project-wide code style guide here: https://www.tensorflow.org/community/contribute/code_style
If these differ, can you reconcile and link here as well?

ricky3dec replied:
Thanks @lamberta. Please let me know, if following text is what you are suggesting?

"Tensorflow project wide code style guidelines can be followed at [TensorFlow Style Guide - Conventions](https://www.tensorflow.org/community/contribute/code_style) 
and Tensorflow I/O project specific code style guidelines can be followed at [Style Guide](STYLE_GUIDE.md)."

lamberta replied:
That works—thanks!

ricky3dec replied:
Updated as suggested! Thanks.

yongtang commented:
Looks like all issues have been resolved so I will just merge. Thanks @ricky3dec for the contribution ,and thanks @MarkDaoust and @lamberta for the kindly help 👍 !
